packages/clay-css/src/scss/atlas/variables/_forms.scss

```diff
@@ -167,92 +167,186 @@ $form-feedback-font-weight: $font-weight-semi-bold !default;
 
 $form-feedback-indicator-margin-x: 0 !default;
 
-// Input Variants
+// .has-error .form-control
 
 $input-danger-bg: $danger-l2 !default;
 $input-danger-border-color: $danger-l1 !default;
 // Will need to be revisited if https://github.com/twbs/bootstrap/pull/24821 merge error is fixed
 $input-danger-box-shadow: 0 0 rgba(0, 0, 0, 0) !default;
-$input-danger-focus-box-shadow: $input-focus-box-shadow !default;
 $input-danger-color: $input-color !default;
 
+$input-danger-focus-box-shadow: $input-focus-box-shadow !default;
+
+$input-danger: () !default;
+$input-danger: map-deep-merge(
+	(
+		background-color: $input-danger-bg,
+		border-color: $input-danger-border-color,
+		box-shadow: $input-danger-box-shadow,
+		color: $input-danger-color,
+		focus: (
+			box-shadow: $input-danger-focus-box-shadow,
+		),
+	),
+	$input-danger
+);
+
+// .has-error .form-control[readonly]
+
 $input-danger-readonly: () !default;
 $input-danger-readonly: map-deep-merge(
 	(
-		bg:
+		background-color:
 			setter(
 				map-get($input-readonly, bg),
 				map-get($input-readonly, background-color)
 			),
 		border-color: map-get($input-readonly, border-color),
-		focus-border-color: $input-focus-border-color,
+		focus: (
+			border-color: $input-focus-border-color,
+		),
```

Comment:
Maybe we have to do what was done with `bg` here? Would we still have to add support for `focus-border-color`?

Reply:
@matuzalemsteles it gets handled by the [clay-form-control-variant mixin](https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/mixins/_forms.scss#L370). The alias is only done here because we are using the `$input-readonly` map directly and not processing it through a mixin.

Reply:
Oh I see, it's clear now, thanks